python/cuml/utils/input_utils.py

```diff
@@ -193,9 +194,20 @@ def convert_dtype(X, to_dtype=np.float32):
             return X_m
 
     elif cuda.is_cuda_array(X):
-        X_m = cp.asarray(X)
-        X_m = X_m.astype(to_dtype)
-        return cuda.as_cuda_array(X_m)
+        try:
+            import cupy as cp
+            X_m = cp.asarray(X)
+            X_m = X_m.astype(to_dtype)
+            return cuda.as_cuda_array(X_m)
+
+        # todo: Remove once CuPy nccl conflicts are dealt with
+        except ImportError:
+            warnings.warn("Using cuDF for dtype conversion, install"
+                          "CuPy for faster data conversion.")
```

Comment:
Should this use a warning category so users can more easily suppress it, possibly a `ImportWarning`?

Followup question: should we mention this, and possibly how to suppress it, in the README?

Reply:
Just talked to @dantegd offline. Since this temporary fix shouldn't extend past 0.8, my questions can be ignored.

Reply:
@rlratzel and @dantegd - are we confident this won't go out in 0.8? I thought it was still an open question about whether cupy packages will be ready in time?

Reply:
> @rlratzel and @dantegd - are we confident this won't go out in 0.8? I thought it was still an open question about whether cupy packages will be ready in time?

I was under the impression that the changes in this PR wouldn't be released with 0.8, but I really can't say for sure. If that changes then I like your suggestions in addition to the warning category and README update.